lib/manageiq/api/common/middleware/web_server_metrics.rb

```diff
@@ -0,0 +1,53 @@
+module ManageIQ
+  module API
+    module Common
+      module Middleware
+        class WebServerMetrics
+          def initialize(app, options = {})
+            @app            = app
+            @metrics_prefix = options[:metrics_prefix] || "http_server"
+
+            require 'prometheus_exporter/client'
+            require 'prometheus_exporter/metric'
+
+            @counter   = PrometheusExporter::Client.default.register(:counter, "#{@metrics_prefix}_total", "The total number of HTTP requests handled by the Rack application.")
+            @gauge     = PrometheusExporter::Client.default.register(:gauge, "#{@metrics_prefix}_busy_threads", "The number of threads currently handling HTTP requests.")
+            @histogram = PrometheusExporter::Client.default.register(:histogram, @metrics_prefix, "The HTTP response duration of the Rack application.")
+          end
+
+          def call(env)
+            @gauge.increment
+
+            start_time = Time.now
```

Comment:
Would it be better to use `Benchmark` here?

Reply:
I would, but I also need the result and didn't want to use an `@` variable

Reply:
If you initialize the `result` local before the block it works.

```
irb(main):005:0> result = nil
=> nil
irb(main):006:0> time = Benchmark.realtime { result = 1 + 1 }
=> 1.8899999645327625e-06
irb(main):007:0> time
=> 1.8899999645327625e-06
irb(main):008:0> result
=> 2
```